osgp/shared/shared/src/main/java/org/opensmartgridplatform/shared/security/providers/EncryptionProvider.java

```diff
@@ -8,13 +8,23 @@
  */
 package org.opensmartgridplatform.shared.security.providers;
 
+import org.opensmartgridplatform.shared.exceptionhandling.EncrypterException;
 import org.opensmartgridplatform.shared.security.EncryptedSecret;
 import org.opensmartgridplatform.shared.security.EncryptionProviderType;
-import org.opensmartgridplatform.shared.security.Secret;
 
 public interface EncryptionProvider {
-    EncryptedSecret encrypt(Secret secret, String keyReference);
-    Secret decrypt(EncryptedSecret secret, String keyReference);
+
+    EncryptedSecret encrypt(byte[] secret, String keyReference) throws EncrypterException;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'org.opensmartgridplatform.shared.exceptionhandling.EncrypterException' which is a runtime exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)